curvefs/src/metaserver/inode_storage.h

```diff
@@ -138,19 +141,45 @@ class InodeStorage {
         uint64_t chunkIndex,
         const S3ChunkInfoList& list2add);
 
-    MetaStatusCode RemoveS3ChunkInfoList(
+    MetaStatusCode DelS3ChunkInfoList(
         std::shared_ptr<StorageTransaction> txn,
         uint32_t fsId,
         uint64_t inodeId,
         uint64_t chunkIndex,
-        uint64_t minChunkId);
+        const S3ChunkInfoList& list2del);
+
+    bool SplitS3ChunkInfoList(const std::string& value,
+                              uint64_t delFirstChunkId,
+                              S3ChunkInfoList* list2add);
 
     std::string RealTablename(TABLE_TYPE type, std::string tablename) {
         std::ostringstream oss;
         oss << type << ":" << tablename;
         return oss.str();
     }
 
+    size_t InodeS3MetaSizeKey(uint32_t fsId, uint64_t inodeId) {
+        std::ostringstream oss;
+        oss << fsId << ":" << inodeId;
+        return Hash(oss.str());
+    }
+
+    bool UpdateInodeS3MetaSize(uint32_t fsId, uint64_t inodeId,
+                               uint64_t size4add, uint64_t size4del) {
+        size_t key = InodeS3MetaSizeKey(fsId, inodeId);
+        uint64_t size = inodeS3MetaSize_[key] + size4add;
+        if (size < size4del) {
+            return false;
+        }
+        inodeS3MetaSize_[key] = size - size4del;
+        return true;
+    }
+
+    uint64_t GetInodeS3MetaSize(uint32_t fsId, uint64_t inodeId) {
```

Comment:
Why not get sChunkInfoList size directly?

Reply:
> Why not get sChunkInfoList size directly?

actually, we had store `s3chunkinfo` by alone, so we can't get it from inode by directly.